src/Documents/Commands/GetDocumentsCommand.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+namespace RavenDB\Documents\Commands;
+
+use RavenDB\Exceptions\IllegalArgumentException;
+use RavenDB\Http\RavenCommand;
+use RavenDB\Http\ServerNode;
+
+class GetDocumentsCommand extends RavenCommand
+{
+    protected array $ids = [];
+    protected array $includes = [];
+    private bool $metadataOnly = false;
+
+    private ?string $startWith = null;
+    private ?string $matches = null;
+    private ?int $start = null;
+    private ?int $pageSize = null;
+    private ?string $exclude = null;
+    private ?string $startAfter = null;
+
+    /**
+     * @throws IllegalArgumentException
+     */
+    public function __construct(array $ids, array $includes, bool $metadataOnly)
+    {
+        parent::__construct(GetDocumentsResult::class);
+
+        if (empty($ids)) {
+            throw new IllegalArgumentException("Please supply at least one id");
+        }
+
+        $this->ids = $ids;
+        $this->includes = $includes;
+        $this->metadataOnly = $metadataOnly;
+    }
+
+    protected function createUrl(ServerNode $serverNode): string
+    {
+        $path = $serverNode->getUrl();
+        $path .= '/databases/';
+        $path .= $serverNode->getDatabase();
+        $path .= '/docs?';
+
+        if ($this->start != null) {
+            $path .= '&start=' . $this->start;
+        }
+
+        if ($this->pageSize != null) {
+            $path .= '&pageSize=' . $this->pageSize;
+        }
+
+        if ($this->metadataOnly) {
+            $path .= '&metadataOnly=true';
+        }
+
+        if ($this->startWith != null) {
+            // todo: find php alternative for escapeDataString
+//            pathBuilder.append("&startsWith=");
+//            pathBuilder.append(UrlUtils.escapeDataString(_startWith));
+//
+            if ($this->matches != null) {
+                $path .= '&matches=' . urlEncode($this->matches);
```

Comment:
you can create and use fake class and then we can change impl:
https://github.com/ravendb/ravendb-jvm-client/blob/fe4373a6b02f65df96b916064c43ce3e4b075aa1/src/main/java/net/ravendb/client/util/UrlUtils.java#L20